ideascube/serveradmin/package.py

```diff
@@ -0,0 +1,57 @@
+
+
+import yaml
+import os.path as op
+import zipfile
+import codecs
+import csv
+
+
+class MediaCenterPackage:
+    def __init__(self, working_dir, medias=None):
+        self.working_dir = working_dir
+        self.medias = list(filter(self.check_media, medias or []))
+
+    def check_media(self, media):
+        for required_metadata in ('title', 'summary', 'credits', 'path'):
+            if not media.get(required_metadata):
+                return False
+        if not op.exists(op.join(self.working_dir, media['path'])):
+            return False
+        preview_path = media.get('preview')
+        if (preview_path
+         and not op.exists(op.join(self.working_dir, preview_path))):
+            print("Warning : preview_path {} defined but file does not exists"
+                  .format(preview_path))
+            media['preview'] = None
```

Comment:
Hum, you will get no warning if you use --dry-run then.

Checking that preview path is valid or not is part of the check. It should not be in the zip creation.
(And do we need to keep an optional metadata if we know it is invalid ?)